daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/TypingSpec.scala

```diff
@@ -848,34 +846,34 @@ class TypingSpec extends WordSpec with TableDrivenPropertyChecks with Matchers {
   }
 
   "reject ill formed type synonym definitions" in {
```

Comment:
Can I recommend splitting this into two test cases? I don't see how the negative ones belong here.

Perhaps "accept well-formed type synonym definitions" and "reject ill-formed type synonym definitions"?

(Wow, "synonym" is hard to type.)

Reply:
I really like the way this test is written. 

I believe that having in the same unit positive and negative test cases that share most of their code improves the confidence.
Positive test cases check the negative tests are not broken and reciprocally. 

If you untie this relationship by writing two different test case, you:
1. make the relationship between the two kind of test less obvious.
2. will naturally share less code
3. make easier for developers that will have to refactor this code to break one of the families of tests. 

One compromise would be to use `FreeSpec` instead of `WordSpec` to build a more fine grain hierarchy of test where both kinds are group together while being separate. However, I do not think it is the scope of this PR to refactor so much this file.  

Changing the title of the test does not really help because this makes the notion of negative/positive become blur.

Reply:
Fair enough, good reasoning. 🙂